### PR TITLE
Add documentation for WorkflowResult

### PR DESCRIPTION
**Pull request by mareklibra**
Fixes: FLPATH-1680

Based on #403

### CODE REVIEW

File: best-practices.md [hunk 1]
```diff
@@ -0,0 +1,55 @@
+# Best practices when creating a workflow
+A workflow should be developed in accordance with the guidelines outlined in the  [serverless workflow definitions][1] documentation.
```
Comment (gciavarrini) [inline best-practices.md:2]: nit: "the  [serverless" two whitespace

File: best-practices.md [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+This document provides a summary of several additional rules and recommendations to ensure smooth integration with other applications, such as the [Backstage Orchestrator][2].
```
Comment (gciavarrini) [inline best-practices.md:4]: >  such as the [Backstage Orchestrator][2].

Is it correct to link workflow-result-schema.json ? From the sentence I expect a link to the orchestrator operator or plugin: can you clarify?
In addition, not sure if `Backstage Orchestrator` is already used: better to use a consistent naming, can you double check?

File: best-practices.md (hunk 1 above)
Comment (gciavarrini) [reply best-practices.md:2]: nit: I think it's better to use "Serverless Workflow definition" (capitol letters) as defined in the provided link.

Comment (mareklibra) [reply best-practices.md:2]: Thank you, both fixed.

File: best-practices.md (hunk 2 above)
Comment (mareklibra) [reply best-practices.md:4]: Good catch, it was a mistake.
We are newly referencing the 1.2 helm chart.

Comment (gciavarrini) [reply best-practices.md]: @masayag WDYT about this line?

**Comment by gciavarrini** [comment]
Thanks for this PR :) 
It LGTM, but I'd like to have a hack from @masayag about the naming.

**Comment by mareklibra** [comment]
Related PR: https://github.com/parodos-dev/parodos-dev.github.io/pull/105

Thanks to the PR 1-5, the changes of this PR will go to the `main` version of the Parodos documentation and will not affect its `1.2` version.

File: best-practices.md [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+
+```
+{
+    "$schema": "http://json-schema.org/draft-07/schema#",
+    "title": "WorkflowResult",
+    "description": "Schema of workflow output",
+    "type": "object",
+    "properties": {
+        "result": {
+            "$ref": "workflow-result-schema.json",
```
Comment (gabriel-farache) [inline best-practices.md:46]: Should not it be `shared/schemas/workflow-result-schema.json`?

Comment (mareklibra) [reply best-practices.md:46]: IMO, it doesn't matter, it's an example only. The locations may vary.

File: best-practices.md [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+  - extensionid: workflow-output-schema
+    outputSchema: schemas/workflow-output-schema.json
+states:
+  - name: ImmediatelyEnd
+     type: inject
+     data:
+       result:
+          completedWith: success
+          message: A human-readable description of the successful status. Or an error.
+          output:
```
Comment (gabriel-farache) [inline best-practices.md:27]: should be outputs :)

Comment (mareklibra) [reply best-practices.md]: Done, thanks :-)

File: best-practices.md (hunk 3 above)
Comment (mareklibra) [reply best-practices.md]: Done

File: best-practices.md (hunk 2 above)
Comment (masayag) [reply best-practices.md]: looks good now.

**Review by masayag — APPROVED** [review]

File: README.md [hunk 5]
```diff
@@ -40,17 +41,19 @@ Follow these steps to successfully add a new workflow:
...
     2. Verify pulling the image [here](https://catalog.redhat.com/software/containers/openshift-serverless-1-tech-preview/logic-swf-builder-rhel8/6483079349c48023fc262858?architecture=amd64&image=65e1a56104e00058ecdd52eb&container-tabs=gti)
 
-
 Note on CI:
 For every PR merged in the workflow directory, a GitHub Action runs an image build to generate manifests, and a new PR is automatically generated in the [serverless-workflows-config repository][3]. The credentials used by the build process are defined as organization level secret, and the content is from a token on the helm repo with an expiry period of 60 days. Currently only the repo owner (rgolangh) can recreate the token. This should be revised.
 
 [1]: https://github.com/serverlessworkflow/specification/blob/main/specification.md
 [2]: https://github.com/apache/incubator-kie-kogito-serverless-operator/
 [3]: https://github.com/parodos-dev/serverless-workflows-config
+[4]: ./best-practices.md
```
Comment (masayag) [inline README.md:59]: @mareklibra pls replace this with a full URL since relative URLs will fail to render in the website.

Comment (mareklibra) [reply README.md]: done

**Review by masayag — APPROVED** [review]

**Pull request merged**